[4.0] Fix numerics

brianteeman opened the pull request:
This PR fixes the bug reported in #35539 **AND** removes the Filter Numeric Terms option and related code as it is not complete and doesnt do anything at all

wilsonge committed:
Tag 4.0.3 stable

brianteeman committed:
[4.0] Fix numerics

As you can see this works as well at fixing the reported bug

brianteeman committed:
indexer

Quy commented on administrator/components/com_finder/src/Indexer/Query.php:
```diff
@@ -1283,11 +1283,6 @@ protected function processString($input, $lang, $mode)
 					continue;
 				}
 
-				if ($params->get('filter_numerics', 0) && $token->numeric)
```
There are 5 instances without `s` in this file.:

https://github.com/joomla/joomla-cms/blob/32f6c81fc2f1d35bec3853f345a93421d93dba6f/administrator/components/com_finder/src/Indexer/Query.php#L1010

brianteeman replied:
even more proof that this code never worked and was never tested by the original submitter and it should just be removed

alikon commented:
i'm lost...
what's the relation between #35339 and this ?

brianteeman commented:
oops wrong PR - updated now

dgrammatiko committed:
Fix Cassiopeia Offline (#35553)

richard67 committed:
Fix wrong since tag coming from upmerge with deploy version (#35551)

wilsonge committed:
Reset to dev

chmst committed:
Adapt modal articles to articles (#35564)

brianteeman committed:
[4.0.x] Update skipto (#35590)

This updates the skipto js to the newly released 4.1.2

### Version 4.1.2
Added aria-busy="true" attribute to menu element when SkipTo is initialized and being updated with new menu items to support validators looking for required menu items for the menu role.
Added the optional aria-controls attribute to button element to reference the id of the menu element as defined in the W3C ARIA Authoring practices for menu button pattern.

### Version 4.1.1
Removed aria-describedby from button, since screen readers read the accesskey information.

Commit:
ZIP extraction helper for Joomla Update (#35388)

beat committed:
Fixes Debug bar not displaying logs by default #35454 (#35455)

Fixes bug #35454 (Debug plugin: Debug bar doesn't display logs and deprecated tabs by default, while setting's default is ON)

brianteeman committed:
filter_numeric doesnt exist

brianteeman committed:
[4.0] Logged in users module (#35589)

If you do not have the track session metadata option enabled in global configuration then the admin module for Logged Users displays an appropriate message.

This PR simply styles the module output correctly to be consistent

regularlabs committed:
[J4] Adds support for custom class on subforms via xml (#35577)

* [J4] Adds support for custom class on subforms via xml

* [J4] Adds support for custom class on subforms via xml

wojsmol committed:
rm cassiopeia/images/system (#35549)

brianteeman committed:
[4.x] Tinymce updates (#35605)

See https://www.tiny.cloud/docs/changelog/ for details

This includes 5.9.0, 5.9.1 and 5.9.2

tecpromotion committed:
typo in filename (#35613)

Bakual committed:
Remove JHELP_ strings for com_config and move the ref_key to the config.xml of the respective component. (#35479)

Co-authored-by: Thomas Hunziker <werbemails@bakual.ch>

dgrammatiko committed:
[4.0] Fix modal data attributes (#35626)

* Update main.php

* Update Bootstrap.php

Bakual committed:
[4.0] Remove JHELP_ strings for com_categories (#35534)

PhilETaylor commented on administrator/components/com_finder/src/Indexer/Indexer.php:
```diff
@@ -956,7 +950,11 @@ protected function addTokensToDb($tokens, $context = '')
 			}
 
-			$db->setQuery($query)->execute();
+			// Only execute the query if there are tokens to insert
```
This change seems to be the same as #35540 ?

Commit:
Category custom fields marked as 'subform' act as if they are not (#35547)

* Subform–only fields cause saving a category to fail.

Part I: “Only Use In Subform” doesn't do anything

* Subform–only fields cause saving a category to fail.

Part II. PlgSystemFields::onContentPrepareForm doesn't actually work for com_categories

Close gh-35543

* Subform–only fields cause saving a category to fail.

Part II. PlgSystemFields::onContentPrepareForm doesn't actually work for com_categories

This should only apply on com_content categories, not
third party categories.

* Invert the if–block order

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Subform–only fields cause saving a category to fail.

Part II. PlgSystemFields::onContentPrepareForm doesn't actually work for com_categories

Okay, it could actually apply for any category,
not just com_content.

* Docblock

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Bakual committed:
[4.0] Replace JHELP_ String with hardcoded ref-key (#35429)

* Replace JHELP_ String with hardcoded ref-key

* Simplifying helpTOC.php so it doesn't need to reverse lookup the JHELP_ strings.

* Remove loading of joomla.ini and messing with language class property

* alpha-sort help strings

* Next batch

* Last batch

* Ooops

Co-authored-by: Thomas Hunziker <werbemails@bakual.ch>

Commit:
[4][CloudFlare Issues] Remove validation of the session by IP address (#35509)

> IMHO the commend of @arnepluhar in #35509 should also count as a successful test. He did it as a GitHub code review. The rules about what constitutes a valid test should expand to GitHub code reviews; these didn't even exist when the rules were decided upon back in the day and they make FAR MORE SENSE than comments filed on a PR!

Github is configured to allow maintainers merge only if all tests are successful.

Thanks all.

Bakual committed:
Language Keys may have a : in it. (#35659)

richard67 committed:
[4.0] Fix undefined constant "GLOB_BRACE" error when using TinyMCE on certain OS like e.g. Alpine Linux (#35636)

* Don't use GLOB_BRACE

* Remove redundant call to "Factory::getLanguage"

* Use Folder class from Filesystem library

brianteeman committed:
[4.0] skipto css - cassiopeia (#35537)

* [4.0] skipto css - cassiopeia

PR for #35211

This is a css change so requires npm ci

To test enable the skip-to plugin for the site

tab to reveal the button and hover over it to display the tooltip

### Before

### After

* lint

dgrammatiko committed:
Update SASS to 1.42 (#35666)

brianteeman committed:
[4.0] remove extra spaces (#35695)

Spotted while checking something else. Can be merged on review

laoneo committed:
Check in media field if image is set (#35556)

richard67 commented:
@brianteeman Now as PR #35540 has been merged, I would expect this PR here to show a conflict because it contains the same change as in PR #35540 . But it doesn't show any conflict on GitHub, and it still shows the change from PR #35540 , too. Could you update your branch to the 4.0-dev of the CMS to solve that? Or if you don't have the time, allow me to do that? I'd like that to be solved before I test.

brianteeman commented:
the change is the same so there wont be a conflict and its why it doesnt show up as a change (now)

richard67 commented:
@brianteeman Did you close it by purpose, and will you make a new one? Maybe I can try to fix our branch, but the easiest thing would be to make a new one, I think.
